hack/testing/rsyslog/viaq_formatting.conf

```diff
@@ -159,16 +159,31 @@ if strlen($!_MACHINE_ID) > 0 then {
         }
     }
     unset $!PRIORITY;
-    if strlen($!message) == 0 then {
-        if strlen($!MESSAGE) > 0 then {
-            set $!message = $!MESSAGE;
+    # rsyslog 8.30.0 and later does case insensitive variable name comparison
+    # which means $!MESSAGE is the same as $!message - HOWEVER - the case
+    # of the variable name is preserved when outputting, so we need to "normalize"
+    # to all lower case so that the internal JSON to string conversion will output
+    # "message" in the outgoing record
+    if (strlen($!message) == 0) and (strlen($!MESSAGE) == 0) then {
+        if strlen($!log) > 0 then {
+            set $!message = $!log;
+        }
+    } else {
+        # see if we're using rsyslog with case sensitivity
+        if $!message == $!MESSAGE then {
+            # in case it is really $!MESSAGE - we have to completely unset it, then
+            # set it in lower case
+            set $.save_message = $!message;
+            unset $!message;
+            set $!message = $.save_message;
+            unset $.save_message;
         } else {
-            if strlen($!log) > 0 then {
-                set $!message = $!log;
+            if strlen($!message) == 0 then {
+                # case sensitive
+                set $!message = $!MESSAGE;
```

Comment:
Could you confirm that we don't have to worry about this case?  $!MESSAGE is just trashed.
`(strlen($!message) > 0) and (strlen($!MESSAGE) > 0) and ($!message != $!MESSAGE)`

Reply:
yes - in this case, `$!message` was somehow set, which means we need to preserve its value.  This case shouldn't happen in this section of code, which deals with the raw data read from journald.